azimuth/modules/dataset_analysis/similarity_analysis.py

```
@@ -45,10 +44,7 @@ def get_encoder_name_or_path(self):
         return model_name_or_path
 
     def get_encoder(self):
```

Should we raise an exception if `not self.can_load_encoder`? I understand it is not possible now since `get_encoder()` is only defined in `FAISSModule`, which has `can_load_encoder = True`, but it would be an extra safety net for if someone tries to create, for some reason, a new module that inherits `FAISSModule` and overwrites `can_load_encoder = False`.